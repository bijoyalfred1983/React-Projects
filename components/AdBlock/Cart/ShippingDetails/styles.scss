@import '../../../theme/variables';

.shippingDetailsContainer {
  background-color: $color-white-main;
  width: 668px;
  margin: auto;
  
  .containerRow {
    display: flex;
  }

  .containerColLeft {
    display: flex;
    padding: 30px 0 30px 16px;
    flex: 1.1;
  }

  .containerColRight {
    display: flex;
    padding: 30px 16px 30px 0;
    flex: 0.9;
  }

  .shipingDetails {
    margin: 0 17px;
    flex: 1;
  }

  .shipImg {
    width: 25px;
    height: 20px;
  }

  .homeImg {
    width: 40px;
    height: 40px;
    margin-top: -8px;
    margin-right: -8px;
  }

  .shipTitle {
    font-family: $font-az-medium;
    font-size: 16px;
    font-weight: 500;
    font-style: normal;
    font-stretch: condensed;
    line-height: normal;
    letter-spacing: normal;

    .shopRunnerImg {
      display: inline-block;
      float: right;
      margin-top: -21px;
      margin-right: -20px;
    }
  }

  .subShipTitle {
    font-family: $font-az-app;
    font-size: 12px;
    font-weight: normal;
    font-style: normal;
    font-stretch: normal;
    letter-spacing: normal;
    line-height: 1.67;
    margin-top: 8px;
  }

  .shipInfo {
    margin-top: 8px;
    font-family: $font-az-app;
    font-size: 12px;
    font-weight: normal;
    font-style: normal;
    font-stretch: normal;
    letter-spacing: normal;
    opacity: 0.6;
    line-height: 1.5;
  }

  .addressContainer {
    display: flex;
    margin-top: 12px;
  }

  .addressText {
    font-family: $font-category-cont;
    font-style: normal;
    font-stretch: normal;
    letter-spacing: normal;
    font-size: 14px;
    line-height: 1.43; 
    font-weight: 500;
    margin-right: 10px;
  }

  .closeAtTime {
    font-family: $font-category-cont;
    font-size: 12px;
    font-weight: 500;
    font-style: normal;
    font-stretch: normal;
    line-height: 1.92;
    letter-spacing: normal;
  }

  .linkContent a:hover {
    text-decoration: underline;
  }

  .linkText {
    font-family: $font-category-cont;
    font-size: 12px;
    font-style: normal;
    font-stretch: normal;
    letter-spacing: normal;
    font-weight: 500;
    line-height: 1.58;
    color: $black-bg-color;
  }

  .linkTextUnderline {
    font-family: $font-az-app;
    font-size: 12px;
    font-weight: normal;
    font-style: normal;
    font-stretch: normal;
    letter-spacing: normal;
    line-height: 1.7;
    color: $black-bg-color;
    text-decoration: underline;
    border: none !important;
    background: none !important;
    padding: 0;
  }

  .linkContent {
    display: flex;
    align-items: center;
    margin-top: 12px;

    .learnMoreShipHome {
      text-transform: uppercase;
    }
  }

  .line {
    width: 1px;
    height: 9px;
    border: solid 0.5px $az-pipe-grey-text;
    margin: 0 8.5px;
  }
}

@media #{$extra-large-and-down} {
  .shippingDetailsContainer {
    display: none;
  }
}

.shipHomeTitle {
  text-transform: uppercase;
}
