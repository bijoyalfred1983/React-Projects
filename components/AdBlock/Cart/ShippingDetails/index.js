/* @flow */
import React from 'react';
import { Link } from 'react-router-dom';
import styles from './styles.scss';

type Props = {
  cartData: Object,
  staticLabelData: Object,
  changeStoreLinkInteraction: () => void // Analytics
};

export default (props: Props) => (
  <div className={styles.shippingDetailsContainer}>
    <div className={styles.containerRow}>
      <div className={styles.containerColLeft}>
        <div>
          <img
            className={styles.homeImg}
            src="/images/az-store.svg"
            alt="store pickup"
          />
        </div>
        <div className={styles.shipingDetails}>
          <div className={styles.shipTitle}>
            {props.staticLabelData.label_cart_bopus_PickUpInStoreSameDay
              ? props.staticLabelData.label_cart_bopus_PickUpInStoreSameDay
              : 'label_cart_bopus_PickUpInStoreSameDay'}
          </div>
          <div className={styles.subShipTitle}>
            {props.staticLabelData
              .label_cart_bopus_SaveTimeBySelectingPickUpInStoreAndWeLlEmailYouWhenProductsAreReady
              ? props.staticLabelData
                  .label_cart_bopus_SaveTimeBySelectingPickUpInStoreAndWeLlEmailYouWhenProductsAreReady
              : 'label_cart_bopus_SaveTimeBySelectingPickUpInStoreAndWeLlEmailYouWhenProductsAreReady'}
          </div>
          <div className={styles.shipInfo}>
            {props.staticLabelData
              .label_cart_bopus_OnlyApplicableIfOrderedMoreThanOneHourBeforeClosingTime
              ? props.staticLabelData
                  .label_cart_bopus_OnlyApplicableIfOrderedMoreThanOneHourBeforeClosingTime
              : 'label_cart_bopus_OnlyApplicableIfOrderedMoreThanOneHourBeforeClosingTime'}
          </div>
          <div className={styles.addressContainer}>
            <div className={styles.addressText}>
              <div>{(props.cartData && props.cartData.storeAddress) || ''}</div>
              <div>
                {`${(props.cartData && props.cartData.storeCity) ||
                  ''} ${(props.cartData && props.cartData.storeState) ||
                  ''} ${(props.cartData && props.cartData.storeZipcode) || ''}`}
              </div>
            </div>
            <div className={styles.linkText}>
              {props.cartData &&
                !props.cartData.open24Hours && (
                  <div className={styles.closeAtTime}>
                    {props.staticLabelData.label_cart_bopus_ClosesAt
                      ? props.staticLabelData.label_cart_bopus_ClosesAt
                      : 'label_cart_bopus_ClosesAt'}
                    {` ${(props.cartData && props.cartData.storeCloseTime) ||
                      ''}`}
                  </div>
                )}
              <button
                className={styles.linkTextUnderline}
                onClick={props.changeStoreLinkInteraction}
              >
                {props.staticLabelData.label_cart_bopus_ChangeStore
                  ? props.staticLabelData.label_cart_bopus_ChangeStore
                  : 'label_cart_bopus_ChangeStore'}
              </button>
            </div>
          </div>
        </div>
      </div>
      <div className={styles.containerColRight}>
        <div>
          <img
            className={styles.shipImg}
            src="/images/shipping-truck.png"
            alt="ship to home"
          />
        </div>
        <div className={styles.shipingDetails}>
          <div className={styles.shipTitle}>
            <div className={styles.shipHomeTitle}>
              {props.staticLabelData.label_cart_shipToHome_ShipToHomeWith
                ? props.staticLabelData.label_cart_shipToHome_ShipToHomeWith
                : 'label_cart_shipToHome_ShipToHomeWith'}
            </div>
            <div className={styles.shopRunnerImg}>
              <img src="/images/shoprunner-logo.png" alt="shoprunner" />
            </div>
          </div>
          {props.cartData && (
            <div className={styles.subShipTitle}>
              {props.staticLabelData
                .label_cart_shipToHome_OrderNowAndGetItBetween
                ? props.staticLabelData
                    .label_cart_shipToHome_OrderNowAndGetItBetween
                : 'label_cart_shipToHome_OrderNowAndGetItBetween'}{' '}
              {`  ${props.cartData.srMinDate || ' '} 
              ${
                props.staticLabelData.label_cart_beforeFooter_And
                  ? props.staticLabelData.label_cart_beforeFooter_And
                  : 'label_cart_beforeFooter_And'
              }
                  ${props.cartData.srMaxDate || ' '} `}
              {props.staticLabelData.label_cart_beforeFooter_With
                ? props.staticLabelData.label_cart_beforeFooter_With
                : 'label_cart_beforeFooter_With'}{' '}
              <strong>
                {props.staticLabelData.label_cart_bopus_Free
                  ? props.staticLabelData.label_cart_bopus_Free
                  : 'label_cart_bopus_Free'}{' '}
              </strong>{' '}
              {props.staticLabelData.label_cart_bopus_2DayShipping
                ? props.staticLabelData.label_cart_bopus_2DayShipping
                : 'label_cart_bopus_2DayShipping'}
              {'.'}
            </div>
          )}
          <div className={styles.linkContent}>
            <Link
              className={`${styles.linkText} ${styles.learnMoreShipHome}`}
              to=""
            >
              {props.staticLabelData.label_cart_shipToHome_LearnMore
                ? props.staticLabelData.label_cart_shipToHome_LearnMore
                : 'label_cart_shipToHome_LearnMore'}
            </Link>
            <div className={styles.line} />
            <Link className={styles.linkText} to="">
              {props.staticLabelData.label_checkout_signin_SignIn
                ? props.staticLabelData.label_checkout_signin_SignIn
                : 'label_checkout_signin_SignIn'}
            </Link>
          </div>
        </div>
      </div>
    </div>
  </div>
);
