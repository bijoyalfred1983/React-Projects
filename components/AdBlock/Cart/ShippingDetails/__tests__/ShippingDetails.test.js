import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import ShippingDetails from '../';
import StaticRouter from 'react-router-dom/StaticRouter';
import { Link } from 'react-router-dom';

describe('<ShippingDetails />', () => {

  let PROPS = {};
  beforeEach(() => {
    const staticLabelData = {"test":"aa"};
    PROPS = {
      cartData: {
        storeZipcode: '38104',
        shopRunnerOn: true,
        storeNextTransitionTime: '7:30 AM',
        storeAddress1: '',
        storeAddress: '385 N Cleveland St',
        srMinDate: '22 Jan',
        srMaxDate: '24 Jan',
        storeState: 'TN',
        storeCity: 'Memphis'
      },
      staticLabelData,
      changeStoreLinkInteraction: jest.fn()
    };
  });

  test('renders correctly ShippingDetails', () => {
    const wrapper = mount(
      <StaticRouter>
        <ShippingDetails {...PROPS}/>
      </StaticRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('check for empty cart data for ShippingDetails', () => {
    const AnalyticsData = {
    "label_cart_bopus_PickUpInStoreSameDay":"a",
    "label_cart_bopus_SaveTimeBySelectingPickUpInStoreAndWeLlEmailYouWhenProductsAreReady":"a",
    "label_cart_bopus_OnlyApplicableIfOrderedMoreThanOneHourBeforeClosingTime":"a",
    "label_cart_bopus_ClosesAt":"a",
    "label_cart_bopus_ChangeStore":"a",
    "label_cart_shipToHome_ShipToHomeWith":"a",
    "label_cart_shipToHome_LearnMore":"a",
    "label_checkout_signin_SignIn":"a",
    "label_cart_bopus_ClosesAt":"a",
    "label_cart_beforeFooter_And":"a",
    "label_cart_beforeFooter_With":"a",
    "label_cart_bopus_2DayShipping":"a",
    "label_cart_bopus_Free":"a"
  }
    PROPS = {
      cartData: {},
      staticLabelData: AnalyticsData
    };
    const wrapper = mount(
      <StaticRouter>
        <ShippingDetails {...PROPS}/>
      </StaticRouter>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  test('test onClick function', () => {
    const wrapper = mount(
      <StaticRouter>
        <ShippingDetails {...PROPS}/>
      </StaticRouter>
    );
      wrapper
      .find(Link)
      .at(0)
      .props()
      .onClick();
    //expect(toJson(wrapper)).toMatchSnapshot();
  });

});
