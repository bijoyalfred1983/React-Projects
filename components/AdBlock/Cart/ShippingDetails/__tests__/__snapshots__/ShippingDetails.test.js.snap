// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ShippingDetails /> check for empty cart data for ShippingDetails 1`] = `
<StaticRouter
  basename=""
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Component
      cartData={Object {}}
      staticLabelData={
        Object {
          "label_cart_beforeFooter_And": "a",
          "label_cart_beforeFooter_With": "a",
          "label_cart_bopus_2DayShipping": "a",
          "label_cart_bopus_ChangeStore": "a",
          "label_cart_bopus_ClosesAt": "a",
          "label_cart_bopus_Free": "a",
          "label_cart_bopus_OnlyApplicableIfOrderedMoreThanOneHourBeforeClosingTime": "a",
          "label_cart_bopus_PickUpInStoreSameDay": "a",
          "label_cart_bopus_SaveTimeBySelectingPickUpInStoreAndWeLlEmailYouWhenProductsAreReady": "a",
          "label_cart_shipToHome_LearnMore": "a",
          "label_cart_shipToHome_ShipToHomeWith": "a",
          "label_checkout_signin_SignIn": "a",
        }
      }
    >
      <div
        className="shippingDetailsContainer"
      >
        <div
          className="containerRow"
        >
          <div
            className="containerColLeft"
          >
            <div>
              <img
                alt="store pickup"
                className="homeImg"
                src="/images/az-store.png"
              />
            </div>
            <div
              className="shipingDetails"
            >
              <div
                className="shipTitle"
              >
                a
              </div>
              <div
                className="subShipTitle"
              >
                a
              </div>
              <div
                className="shipInfo"
              >
                a
              </div>
              <div
                className="addressContainer"
              >
                <div
                  className="addressText"
                >
                  <div />
                  <div />
                  <div>
                    a
                     
                  </div>
                </div>
                <div
                  className="linkText"
                >
                  <div>
                    a
                  </div>
                  <Link
                    className="linkTextUnderline"
                    replace={false}
                    to=""
                  >
                    <a
                      className="linkTextUnderline"
                      href="/"
                      onClick={[Function]}
                    >
                      a
                    </a>
                  </Link>
                </div>
              </div>
            </div>
          </div>
          <div
            className="containerColRight"
          >
            <div>
              <img
                alt="ship to home"
                className="shipImg"
                src="/images/shipping-truck.png"
              />
            </div>
            <div
              className="shipingDetails"
            >
              <div
                className="shipTitle"
              >
                a
              </div>
              <div
                className="subShipTitle"
              >
                label_cart_shipToHome_OrderNowAndGetItBetween
                 
                    
              a
                    
                a
                 
                <strong>
                  a
                   
                </strong>
                 
                a
                .
              </div>
              <div
                className="linkContent"
              >
                <Link
                  className="linkText"
                  replace={false}
                  to=""
                >
                  <a
                    className="linkText"
                    href="/"
                    onClick={[Function]}
                  >
                    a
                  </a>
                </Link>
                <div
                  className="line"
                />
                <Link
                  className="linkText"
                  replace={false}
                  to=""
                >
                  <a
                    className="linkText"
                    href="/"
                    onClick={[Function]}
                  >
                    a
                  </a>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Component>
  </Router>
</StaticRouter>
`;

exports[`<ShippingDetails /> renders correctly ShippingDetails 1`] = `
<StaticRouter
  basename=""
  location="/"
>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Component
      cartData={
        Object {
          "shopRunnerOn": true,
          "srMaxDate": "24 Jan",
          "srMinDate": "22 Jan",
          "storeAddress": "385 N Cleveland St",
          "storeAddress1": "",
          "storeCity": "Memphis",
          "storeNextTransitionTime": "7:30 AM",
          "storeState": "TN",
          "storeZipcode": "38104",
        }
      }
      changeStoreLinkInteraction={[MockFunction]}
      staticLabelData={
        Object {
          "test": "aa",
        }
      }
    >
      <div
        className="shippingDetailsContainer"
      >
        <div
          className="containerRow"
        >
          <div
            className="containerColLeft"
          >
            <div>
              <img
                alt="store pickup"
                className="homeImg"
                src="/images/az-store.png"
              />
            </div>
            <div
              className="shipingDetails"
            >
              <div
                className="shipTitle"
              >
                label_cart_bopus_PickUpInStoreSameDay
              </div>
              <div
                className="subShipTitle"
              >
                label_cart_bopus_SaveTimeBySelectingPickUpInStoreAndWeLlEmailYouWhenProductsAreReady
              </div>
              <div
                className="shipInfo"
              >
                label_cart_bopus_OnlyApplicableIfOrderedMoreThanOneHourBeforeClosingTime
              </div>
              <div
                className="addressContainer"
              >
                <div
                  className="addressText"
                >
                  <div>
                    385 N Cleveland St
                  </div>
                  <div />
                  <div>
                    label_cart_bopus_ClosesAt
                     
                  </div>
                </div>
                <div
                  className="linkText"
                >
                  <div>
                    label_cart_bopus_ClosesAt
                  </div>
                  <Link
                    className="linkTextUnderline"
                    onClick={[MockFunction]}
                    replace={false}
                    to=""
                  >
                    <a
                      className="linkTextUnderline"
                      href="/"
                      onClick={[Function]}
                    >
                      label_cart_bopus_ChangeStore
                    </a>
                  </Link>
                </div>
              </div>
            </div>
          </div>
          <div
            className="containerColRight"
          >
            <div>
              <img
                alt="ship to home"
                className="shipImg"
                src="/images/shipping-truck.png"
              />
            </div>
            <div
              className="shipingDetails"
            >
              <div
                className="shipTitle"
              >
                label_cart_shipToHome_ShipToHomeWith
              </div>
              <div
                className="subShipTitle"
              >
                label_cart_shipToHome_OrderNowAndGetItBetween
                 
                  22 Jan 
              label_cart_beforeFooter_And
                  24 Jan 
                label_cart_beforeFooter_With
                 
                <strong>
                  label_cart_bopus_Free
                   
                </strong>
                 
                label_cart_bopus_2DayShipping
                .
              </div>
              <div
                className="linkContent"
              >
                <Link
                  className="linkText"
                  replace={false}
                  to=""
                >
                  <a
                    className="linkText"
                    href="/"
                    onClick={[Function]}
                  >
                    label_cart_shipToHome_LearnMore
                  </a>
                </Link>
                <div
                  className="line"
                />
                <Link
                  className="linkText"
                  replace={false}
                  to=""
                >
                  <a
                    className="linkText"
                    href="/"
                    onClick={[Function]}
                  >
                    label_checkout_signin_SignIn
                  </a>
                </Link>
              </div>
            </div>
          </div>
        </div>
      </div>
    </Component>
  </Router>
</StaticRouter>
`;
